More unnecessary if drops

BBosman opened the pull request:
Both `proj` and `lazyEvaluator` are local variables which get instantiated earlier in the functions guaranteeing them to be non-`null`, making the checks unnecessary.

In the `proj` case there are a couple of references to `proj.ProjectName` before the check, proving the point. :smile:

cdmihai commented on src/XMakeBuildEngine/Evaluation/Evaluator.cs:
```diff
@@ -825,28 +825,24 @@ private static ProjectTargetInstance ReadNewTargetElement(ProjectTargetElement t
             }
 
-            if (lazyEvaluator != null)
+            // Tell the lazy evaluator to compute the items and add them to _data
```
The lazy evaluator was previously introduced and the null / local instantiation are a sort of a poor man's escape hatch to revert back to the old behaviour if need be. At some point in the future (after the next VS release maybe?) we should remove the escape hatch and delete all the downstream code.

cdmihai commented on src/OrcasEngine/Shared/SolutionParser.cs:
```diff
@@ -594,16 +594,13 @@ private void ParseProject(string firstLine)
                 new BuildEventFileInfo(SolutionFile), "SolutionParseProjectEofError", proj.ProjectName);
 
-            if (proj != null)
+            // Add the project to the collection
```
see https://github.com/Microsoft/msbuild/pull/963/files#r76450977

cdmihai commented:
I'm fine with the change in `src/XMakeBuildEngine/Construction/Solution/SolutionFile.cs`. Can you please extract it out in a separate PR so it can be merged? Or we can wait until the rest of the diffs are discussed.

BBosman commented:
I've updated the PR to remove the `src/OrcasEngine` change from it and I've moved the `src/XMakeBuildEngine/Construction/Solution/SolutionFile.cs` change to its own PR (#971).

cdmihai commented:
Closing as this change will become obsolete once we decide to do #975
